modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/config/DefaultImplicitConfigurationProviderFactory.java

```diff
@@ -113,4 +119,47 @@ private boolean needsDynamicConnectionProvider(ExtensionModel extensionModel, Co
     return !getConnectedComponents(extensionModel, configurationModel).isEmpty()
         && implicitConnectionProviderValueResolver.isDynamic();
   }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public String resolveImplicitConfigurationProviderName(ExtensionModel extensionModel, ConfigurationModel configurationModel,
+                                                         MuleContext muleContext) {
+    if (muleContext.getArtifactType().equals(POLICY)) {
+      return featureFlaggedPolicyExtensionsImplicitConfigurationProviderName(extensionModel, configurationModel, muleContext);
+    } else {
+      return getImplicitConfigurationProviderName(extensionModel, configurationModel);
+    }
+  }
+
+  /**
+   * Resolves the {@link ConfigurationProvider} name for a policy extension, taking into account the
+   * {@link MuleRuntimeFeature#ENABLE_POLICY_ISOLATION} feature flag.
+   * 
+   * @param extensionModel     The configurable {@link ExtensionModel}.
+   * @param configurationModel The {@link ConfigurationModel} that represents the extensionModel configuration.
+   * @param muleContext        The corresponding {@link MuleContext}.
+   * @return The {@link ConfigurationProvider} name.
+   */
+  private String featureFlaggedPolicyExtensionsImplicitConfigurationProviderName(ExtensionModel extensionModel,
+                                                                                 ConfigurationModel configurationModel,
+                                                                                 MuleContext muleContext) {
```

Comment:
have the mule context injected into this class instead of receiving it as parameter

Reply:
review all the chained calls where the mule context is being passed around